gemrb/GUIScripts/iwd/CharGen.py

```diff
@@ -535,49 +555,41 @@ def SetCharacterDescription():
 					TextArea.Append ("+")
 					j = j + 1
 				TextArea.Append ("\n")
+	return
 
-		if MageSpell !="*":
-			TextArea.Append ("\n")
-			TextArea.Append (11027)
-			TextArea.Append (":\n")
-			t = GemRB.GetPlayerStat (MyChar, IE_ALIGNMENT)
-			Learnable = Spellbook.GetLearnableMageSpells (GemRB.GetPlayerStat (MyChar, IE_KIT), t,1)
-			MageSpellBook = GemRB.GetVar ("MageSpellBook")
-			MageMemorized = GemRB.GetVar ("MageMemorized")
-			for i in range (len(Learnable)):
-				if (1 << i) & MageSpellBook:
-					Spell = GemRB.GetSpell (Learnable[i])
-					TextArea.Append (Spell["SpellName"])
-					if (1 << i) & MageMemorized:
-						TextArea.Append (" +")
-					TextArea.Append ("\n")
+def GetClassFlag(TableName):
+	if TableName == "MXSPLPRS" or TableName == "MXSPLPAL":
+		return 0x8000
+	elif TableName == "MXSPLDRU":
+		return 0x4000
+	elif TableName == "MXSPLRAN":
+		return 0x4000
+	else:
+		return 0
 
-		if PriestSpell == "*":
-			PriestSpell = DruidSpell
-		if PriestSpell!="*":
-			TextArea.Append ("\n")
-			TextArea.Append (11028)
-			TextArea.Append (":\n")
-			t = GemRB.GetPlayerStat (MyChar, IE_ALIGNMENT)
-			if PriestSpell == "MXSPLPRS" or PriestSpell == "MXSPLPAL":
-				ClassFlag = 0x4000
-			elif PriestSpell == "MXSPLDRU" or PriestSpell == "MXSPLRAN":
-				ClassFlag = 0x8000
-			else:
-				ClassFlag = 0
+def GetLearnableSpellInfo(Learnable, SpellBook, SpellToMark = 0):
+	info = ""
+	for i, SpellRef in enumerate (Learnable):
+		if (1 << i) & SpellBook:
+			Spell = GemRB.GetSpell (SpellRef)
+			info += GemRB.GetString (Spell["SpellName"])
+			if (1 << i) & SpellToMark:
+				info += " +"
+			info += "\n"
+	return info
 
-			Learnable = Spellbook.GetLearnablePriestSpells( ClassFlag, t, 1)
-			PriestMemorized = GemRB.GetVar ("PriestMemorized")
-			for i in range (len(Learnable)):
-				if (1 << i) & PriestMemorized:
-					Spell = GemRB.GetSpell (Learnable[i])
-					TextArea.Append (Spell["SpellName"])
-					TextArea.Append (" +\n")
-	return
 
+def GetKnownSpellsInfo(SpellType):
```

Comment:
If this is identical to what is used in bg1, move it to Spellbook.py and add a comment that it's for the chargen spell listing display.

Reply:
It could also handle the + for memorized spells if needed.

Reply:
Done that. I've reversed order in which spells are listed, because that's what IWD and BG games are doing.

Reply:
bg1 does it differently or can we use it there too?